changelog/pending/20250203--programgen--add-conformance-test-for-dynamic-lists-and-returns.yaml

```diff
@@ -0,0 +1,4 @@
+changes:
+- type: feat
+  scope: programgen
+  description: Add conformance test for dynamic lists and returns
```

Comment:
Nit: Generally, I only include a changelog when there are user facing changes (a fix or improvement). When a change is only adding tests, I personally don't think we need a changelog entry (label the PR with `impact/no-changelog-required`). Curious what others think.

Fine to leave it on this one to avoid the extra CI run.

Reply:
Ack, I wouldn't add a changelog for this